src/test/plugins/mvcc_delete_plugin_system_test.cpp

```diff
@@ -0,0 +1,250 @@
+
+  // This thread calls update_next_row() continuously. As a PausableLoopThread, it gets terminated together with the
+  // test.
+  auto table_update_thread =
+      std::make_unique<PausableLoopThread>(std::chrono::milliseconds(10), [&](size_t) { update_next_row(); });
+
+  // (3.2) Wait until the thread has finished invalidating rows in chunk two
+  while (_counter < CHUNK_SIZE * 2) {  // -> if(_counter < 400)...
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  }
+
+  // (4) Wait for the MvccDeletePlugin to delete chunk two logically
+  {
+    auto attempts_remaining = 10;
+    while (attempts_remaining--) {
+      // The second chunk should have been logically deleted by now
+      if (_table->get_chunk(ChunkID{1})->get_cleanup_commit_id()) break;
+
+      // Not yet. Give the plugin some more time.
+      std::this_thread::sleep_for(MvccDeletePlugin::IDLE_DELAY_LOGICAL_DELETE);
+    }
+    // Check that we have not given up
+    EXPECT_GT(attempts_remaining, -1);
+  }
+
+  // (5) Verify the correctness of the logical deletion operation.
+
+  // Updates started from row 220 on. So chunk two contained 20 rows still valid before its logical deletion.
+  // These rows must have been invalidated and reinserted to the table during the logical delete operation
+  // by the MvccDeletePlugin.
+  validate_table();
+
+  // (6) Wait for the MvccDeletePlugin to delete chunk two physically
+
+  some_other_transaction_context = nullptr;
+  {
+    auto attempts_remaining = 10;
+    while (attempts_remaining--) {
+      // The second chunk should have been physically deleted by now
+      if (_table->get_chunk(ChunkID{1}) == nullptr) break;
+
+      // Not yet. Give the plugin some more time.
+      std::this_thread::sleep_for(MvccDeletePlugin::IDLE_DELAY_PHYSICAL_DELETE);
+    }
+
+    // Check that we have not given up
+    EXPECT_GT(attempts_remaining, -1);
+  }
+
+  // (7) Check after conditions
+
+  validate_table();
+
+  // The first chunk was never modified, so it should not have been cleaned up
+  EXPECT_FALSE(_table->get_chunk(ChunkID{0})->get_cleanup_commit_id());
+
+  // The third chunk was the last to be modified, so it should not have been cleaned up either. (compare criterion 2)
+  EXPECT_FALSE(_table->get_chunk(ChunkID{2})->get_cleanup_commit_id());
+
+  // (8) Prepare clean-up of chunk three
+
+  // Kill a couple of commit IDs so that the third chunk is eligible for clean-up too. (so criterion 2 is fulfilled)
+  for (auto transaction_idx = CommitID{0}; transaction_idx < MvccDeletePlugin::DELETE_THRESHOLD_LAST_COMMIT;
+       ++transaction_idx) {
+    TransactionManager::get().new_transaction_context()->commit();
+  }
+
+  // (9) Wait for the MvccDeletePlugin to delete chunk three logically
+  {
+    auto attempts_remaining = 10;
+    while (attempts_remaining--) {
+      // The second chunk should have been logically deleted by now
+      if (_table->get_chunk(ChunkID{2})->get_cleanup_commit_id()) break;
```

Comment:
@julianmenzler Who guarantees that chunk 2 is not physically deleted as well? There should be a transaction that still to which the chunk is still visible. After step 9, we commit that transaction, so that the physical delete can run.

Reply:
I think you are right. The physical delete might happen too early. We have no guarantee to see the cleanup-commit-id of the logical delete. 
I'll remove the logical-delete-checks, so that we only check for the end result, the physical delete.

Reply:
No need to remove it, just hold a tx

Reply:
I just saw our "blocker condition" for the first physical delete. Therefore, we should be able to check for the cleanup-commit-id of the first logical delete. But the second logical delete cannot be checked.

Reply:
Why not? If we create a transaction before step 10 and hold it until we have verified the logical delete, wouldn't the physical delete be blocked?

Reply:
My last comment still appears unaddressed.

Reply:
Now it is.